mantidproject/mantid
# BASISReduction: optional normalization by proton charge

**Description of work.**
- [x] source for normalization by proton charge.
- [x] source for normalization by duration.
- [x] updated algorithm documentation.
- [x] updated release notes.
- [x] test for reviewer

**Report to:** nobody

**To test:**

Run the following three reductions with different flux normalization types and verify that the resulting spectra are very similar. (Note: run `90177` had a duration of 24 hours but one of those hours had no neutrons. Thus, 
spectra `BSS_90177_sqw_Dur` obtained using the run duration will be most dissimilar)

```
for flux_normalization_type in ('Monitor', 'Proton Charge', 'Duration'):
    BASISReduction(RunNumbers='90177',
        DoFluxNormalization=True,
        FluxNormalizationType=flux_normalization_type,
        ReflectionType='silicon_111',
        EnergyBins=[-100, 0.4, 100],
        MomentumTransferBins=[0.3, 0.2, 1.9])
    RenameWorkspace('BSS_90177_sqw', 'BSS_90177_sqw_'+flux_normalization_type[0:3])
```


Fixes #25348

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
